Sprint 22(home work) Gr26

amanzhola opened the pull request.

amanzhola committed:
Sprint 21 (home work) Gr26

as per review:
+ build.gradle.kts [clean from comments as per Anastasia review/в комментах были подсказки на получение зависимости/]
+ list_item_history.xml(related to Movie-> HistoryViewHolder)[updated using resource for dimension 8 dp as per Anastasia review]

Total update for sprint 21 ->
Play Tracks:

update
build.gradle.kts (add Room, Kapt)->[clean from comments as per Anastasya review]
libs.versions.toml (add Room)
AndroidManifest(cleaning)
AudioRepositoryImpl(Получаем ID избранных треков из базы и Проставляем isFavorite для избранных)
ExtraOptionFragment(transfer from activity to fragment)
ExtraOptionViewModel(+ интерактор для избранного)
extraOptionViewModelModule(DI update)
FavoritesRepositoryImpl(new for Favorite)
FavoriteTrackDao(new for data base)
FavoriteTrackDatabase(new for data base)
FavoriteTrackEntity(new for data base)
FavoriteTracksInteractor (new for Favorite)
FavoriteTracksRepository (new for Favorite)
FavoriteTracksState (new for FavoriteTracksViewModel)
FavoriteTracksViewModel (update for FragmentFavouriteTracks)
fragment_favourite_tracks.xml(add recycle)
FragmentFavouriteTracks(updated to show favorite tracks)
MediaViewModelModule(update DI)
playlist_navigation_graph.xml(Global from FragmentFavouriteTracks for ExtraOptionFragment)
searchDataModule(DI update)
SearchHistoryInteraction(update replace by suspend)
SearchHistoryInteractionImpl(update replace by syspend)
SearchHistoryRepository(update replace by syspend)
SearchRepositoryModule(DI update)
SearchViewModel(+ changed to flow)
Track (added Favorite bool)
TrackAdapter (added actionId: for transfer in Fragments)
TrackAdapterAudio (action for Favorite button)
TrackPreviewActivity (reflecting Track Model changes)

Base:
SharedPreferencesSearchHistoryRepository(update for History to Favorite data base [для Истории Треков аудио])

Movie:

adding data base for movies shown last on search movie screen

updated->
AboutFragment, AboutViewModel, activity_movie_pager.xml, activity_movie_pager_list.xml, DetailsFragment, DetailsViewPagerAdapter, fragment_info.xml, fragment_names.xml, i_movie1.xml, InfoFragment, interface.kt, Movie, movie_navigation_graph.xml, movie_navigation_items.xml, MovieDetailsViewModel, MoviePager, MoviePagerList, movieRepositoryModule, MovieRootActivity, MoviesCastAdapterDelegates.kt, MoviesCastFragment, MoviesCastRVItem, MoviesCastState, MoviesCastViewModel, MovieSearchDto, MoviesRepositoryImpl, MoviesRepositoryImplPoster, movieViewModelModule, NamesFragment, namesInteractorModule, NamesState, NamesViewModel, NavigationHost, PersonsAdapter, PersonViewHolder, PosterFragment, PosterViewModel, SearchMovie,string( + string(en) for adding note on empty screen)

new ->
fragment_history.xml, HistoryAdapter, HistoryFragment, HistoryInteractor, HistoryInteractorImpl, HistoryState, HistoryViewHolder, list_item_history.xml(for HistoryViewHolder)[updated using resource for dimension 8 dp as per Anastasia review], HistoryViewModel, ic_history.xml, MovieDao, MovieEntity, MoviesDatabase

LauncherActivity(update previous not related this data base and favorite updates)

amanzhola committed:
Sprint 21 (home work) Gr26

Risk Management: Removing LiveData & init from VM @coroutine ->
updates:
1)AboutViewModel + AboutFragment  + AboutState
2)MovieDetailsViewModel + DetailsFragment
3)ExtraOptionViewModel + ExtraOptionFragment
4)FavoriteTracksViewModel + FragmentFavouriteTracks + FavoriteTracksState
5) MoviesCastViewModel + MoviesCastFragment + MoviesCastState
6) PosterViewModel + PosterFragment + DI(MovieViewModelModule)
7) HistoryViewModel + HistoryFragment
8) NamesViewModel + NamesFragment
9) MoviesViewModel + SearchMovie
10) ExtraOptionViewModel + ExtraOptionFragment
11) SearchViewModel + SearchFragment.kt + SearchUiState + UIUpdater

12) update:
movie_navigation_graph.xml(update address movie class re-location)

13) fix movie history(History Fragment and it's VM) ->
updates:
1. data
movie_db -> MovieDao
repository -> MoviesRepositoryImpl
2. domain
impl -> movie_db -> HistoryRepositoryImpl

14) Day and Night mode ->
updates:
list_item_history.xml
fragment_history.xml

15) +ClearButton for Edit ->
updated:
activity_search_movie.xml

16) Fix on search movie ->
updated:
SearchMovie (fix btn, showhistory etc)

amanzhola committed:
Sprint 21 (home work) Gr26

1. Fix favorite button reaction:
Update logic -> ExtraOptionViewModel
Update ui -> ExtraOptionFragment
2. Fix onSearchActionDone() for btnUpdate.setOnClickListener and states logic -> SearchViewModel

amanzhola committed:
Sprint 22(home work) Gr26

updated:
1. add_photo.xml пунктир экран для фото
2. AddTrackToPlaylistUseCase для VM аудио
3. App обновления DI
4. bottom_sheet_bg.xml округления BS
5. btn_create_bg.xml селетор для кнопки выбора CreatePlaylistFragment
6. bottom_sheet_bg.xml фон для BS
7. cover_create_playlist.xml картинка для обложки CreatePlaylistFragment
8. DI module CreatePlaylistFragment and VM
9. CreatePlaylistFragment - ui для создания альбомов
10. CreatePlaylistUseCase для VM
11. CreatePlaylistViewModel - VM
12. databaseModule - для DB
13. deploymentTargetSelector.xml - IDEA
14. dimens.xml для sw320dp, sw360dp, sw393dp экранов CreatePlaylistFragment
15 edit_box_outline_selector.xml обводка полей ввода
16. ExtraOptionFragment ui BS
17. ExtraOptionViewModel логика BS
18. extraOptionViewModelModule -> DI
19. FileCopier для коирования пути сохранения для CreatePlaylistViewModel
20. FavoriteTrackDatabase для BD фаворит песен
21. fragment_create_playlist.xml -> xml
22. fragment_playlist.xml -> xml
23. fragment_search.xml -> только в style убрана всплывающая подсказка
24. FragmentPlaylist -> ui playlists
25. GlideImageLoader -> загрузка фото
26. hint_color_selector.xml -> селектор обвода поля ввода CreatePlaylistFragment
27. ImageLoader загрузка фото
28. imageLoaderModule -> util DI
29. item_playlist.xml -> карты для ресайкла сеточного CreatePlaylistFragment
30. item_playlist_bottom.xml -> карты для ресайкла BS
31. MainActivity -> dialog для выхода из приложения
32. mediaViewModelModule -> создание отдельно DI module for PlayList
33. MovieRootActivity -> dialog for exit
34. MoviesRepository.kt -> удален дублирование
35. MoviesViewModel -> упрощена логика с гибкого на деклативный стиль варинт избежать мигания резов предыдущего поиска (более грубый вариант)
36. ObservePlaylistsUseCase -> просморт списка для VM audioPlayer(extraOption)
37. playlist_navigation_graph.xml -> добавление переходов
38. PlaylistBottomAdapter -> BS
39. PlaylistDao -> db
40. PlaylistEntity -> для db
41. PlaylistGridAdapter -> ui сетка для Playlist Fragment
42. PlaylistMappers.kt -> избавление от зависимости
43. PlaylistRepository -> + функции и удаление дубля в Create
44. PlaylistRepositoryImpl -> impl и удаление дубля в Create
45. playlistsModule -> DI
46. PlaylistTrackDao -> new DB
47. PlaylistTrackEntity -> new for DB
48. PlaylistViewModel -> актуализация списков из BD
49. rounder_bsh.xml -> для ui BS
50. SearchMovie -> адаптация изменения VM
51. SpacesItemDecoration -> FragmentPlaylist отступы при сетке
52. strings.xml + string resources
53. themes.xml -> удаление всплывающей подсказки переход на поле ввода с его коробки
54. track_item1.xml - одиночный хмл для аудио, ничего не сделано было ошибочная попятка в нем делать BS из-за вертик\горизонт скролов
55. TrackAdapterAudio добавлена активация кнопки добавления в создание альбомов
56. TrackPreviewActivity заглушка для TrackAdapterAudio кнопки добавления в создание альбомов

anastasianadelshtehel commented on app/src/main/java/com/example/playlistmaker/ui/mediaFragments/PlaylistGridAdapter.kt:
```diff
@@ -0,0 +1,55 @@
+
+    private fun pluralizeTracks(ctx: Context, count: Int): String {
+        // может быть plurals. Простой вариант:
+        return if (count == 1) "1 трек" else "$count треков"
```
Строки нужно выносить в ресурсы

anastasianadelshtehel commented on app/src/main/java/com/example/playlistmaker/ui/createPlaylist/CreatePlaylistFragment.kt:
```diff
@@ -0,0 +1,212 @@
+
+    override fun onResume() {
+        super.onResume()
+        Log.i("CreatePlaylistUI", "onResume; vm=${vm.hashCode()}")
```
Логирование нужно всегда удалять

anastasianadelshtehel commented on app/src/main/java/com/example/playlistmaker/ui/mediaFragments/FragmentPlaylist.kt:
```diff
@@ -42,6 +61,57 @@ class FragmentPlaylist : Fragment() {
+        }
+
+        val handle = findNavController().currentBackStackEntry?.savedStateHandle
+        handle?.getLiveData<String>("playlist_created_name")
```
Ключ можно вынести  в константы

amanzhola committed:
Sprint 22(home work) Gr26
updated(Спасибо Анастисии за классные ревьюшки):
1.
Exception failure caused by BS ->
ExtraOptionFragment:
BottomSheetBehavior.STATE_DRAGGING,
BottomSheetBehavior.STATE_SETTLING
Перед отправкой поставил TODO, обычно не падает, но получается в BS падает
Заменил комментом -> альфой рулит onSlide()

2. CreatePlaylistFragment: Constants.kt for NavKeys -> PLAYLIST_CREATED_NAME

3. FragmentPlaylist: handle -> Constants.kt for NavKeys -> PLAYLIST_CREATED_NAME

4. PlaylistGridAdapter: string resources moved to plurals (strings.xml: rus and en)

5. Constants.kt updated with key

6. Removed underlined of cards в PlaylistGridAdapter:
item_playlist.xml -> strokeColor, cardBackgroundColor, rippleColor

7. Logs cleaned: добавочно +
CreatePlaylistViewModel, PlaylistRepositoryImpl, PlaylistViewModel

PS:
1. всплывающие сообщения через Snackbar на 4 секунды
2. Обложка в CreatePlaylistFragment для sw320dp and sw360dp меньше для сохранения квадратности (TЗ) и поля ввода чуть больше обложки по ТЗ
sw393dp и далее совпадает за счет работы хмл ресурса и выглядит красивее поэтому оставил ширину обложки на усмотрение самого ресурса

amanzhola committed:
Sprint 22(home work) Gr26
1. playlistsModule logs cleaning
2. Previous commenting:
«Прошу прощения за опечатку в имени Анастасии.»

amanzhola committed:
Sprint 22(home work) Gr26
1. playlistsModule logs cleaning
2. Previous commenting:
«Прошу прощения за опечатку в имени Анастасии.»

amanzhola committed:
Sprint 22(home work) Gr26

1. dimens.xml:
Максимальный возможный компромисс для сохранения квадратности обложки Create и ТЗ для sw320dp, sw360dp -> 0.85, sw320dp -> 0.9, выше -> 0.95 для layout_constraintWidth_percent
2. fragment_create_playlist.xml -> убрана связка низа с полем ввода что могло ограничить ширину обложки для формирования квадрата

amanzhola committed:
Sprint 22(home work) Gr26

1. Исправлен рассинхрон кнопок нижнего навигатора при переключении темы в аудио и возврате на поиск. Показывает 2ю 3ку-кнопок вместо первой 3ки [поиск в 1ой тройке], исправляется при повторном переключении [источник рассинхрон], делалось один раз в BaseFragment
-> сделана одна точка "правды", убран ручное регулирование ->
BaseFragment, BottomNavigationHelper, MainActivity
(Ранее иногда замечал, но не мог поймать в тестах, помог переход из поиска в create при создании альбома и после переключения темы возврат автоматом в аудио и далее возврат в поиск и устойчивость ошибки помогла поймать)

2. Исправлена центрование по горизонтали крестика очистки поля ввода (побочка убирания всплывающей подсказки)
Пытался через встроеный,через свой хмл, через фрагмент - ноль результатов
Когда сдался и вернул все как было -> само по себе разрешилось (пока не понятно как исправилась: сама по ходу покосилась и сама же исправилась)

Поэтому коды затем в мусор в fragment_search.xml(черех хмл), SearchFragment(через код), themes.xml(через стиль)

3.dimens.xml -> для перестраховки поставлен 0.75 для самого маленького экрана sw320dp для места под квадрат обложки Create и для пространства между полем ввода и нижней кнопкой ввода

amanzhola committed:
Sprint 22(home work) Gr26

1. Испрвление "исправления" рассинхрона кнопок ниж навигатора, прежнее убила 2ю 3ку кнопок (без перехода на 2ю 3ку) потому вернул все прежние коды в BottomNavigationHelper и MainActivity, BaseFragment -> только в BaseFragment if (!wasBottomNavSynced)  вызываем main.bottomNavigationHelper.setBottomNavigationVisibility()

2. Доп: При тестах ниж навиг увидел и исправил цвет ночной темы для тулбара в CreatePlaylistFragment

amanzhola committed:
Sprint 22(home work) Gr26
🙏 ➡️ 🔥 💃 (👀💎!✨)

1. Исправлена отсутсвие округления обложек в FragmentPlaylist
Округление верхних углов 😌 случайное совпадение: MaterialCardView скругляет саму карточку,а обложка (ImageView) занимает только верхний квадрат (над текстом) и не клипается по радиусу карточки. Поэтому визуально «совпали» только верхние углы.

➡️ item_playlist.xml ➕ rounded_8dp.xml (углы обложки округлены)
-> работа отдана ресурсам

2. Отключена восстановление скролла, чтобы при возврате из CreatePlaylistFragment в FragmentPlaylist скролл ресайклера всегда был на верху списка

➡️ fragment_playlist.xml (❌ RecyclerView)(android:saveEnabled="false")
-> работа отдана ресурсам

3. ➕ 🧰  emoji в strings.xml
